src/ctds/cursor.c

```diff
@@ -469,49 +491,61 @@ static PyObject* create_column_description(const struct Column* column)
     return NULL;
 }
 
-static PyObject* Cursor_description_get(PyObject* self, void* closure)
+static PyObject* ResultSetDescription_get_object(struct ResultSetDescription* description)
 {
-    PyObject* description = NULL;
-
-    struct Cursor* cursor = (struct Cursor*)self;
-
-    /*
-        Return None if the resultset is empty (i.e. has no columns).
-        This will also cover cases where no operation has occurred yet.
-    */
-    if (!cursor->description)
-    {
-        Py_RETURN_NONE;
-    }
-
-    description = PyTuple_New((Py_ssize_t)cursor->description->ncolumns);
-    if (description)
+    if (!description->_obj)
     {
-        size_t ix;
-        for (ix = 0; ix < cursor->description->ncolumns; ++ix)
+        PyObject* tuple = PyTuple_New((Py_ssize_t)description->ncolumns);
+        if (tuple)
         {
-            PyObject* column = create_column_description(&cursor->description->columns[ix]);
-            if (column)
+            size_t ix;
+            for (ix = 0; ix < description->ncolumns; ++ix)
             {
-                PyTuple_SET_ITEM(description, (Py_ssize_t)ix, column); /* column reference stolen by PyTuple_SET_ITEM */
+                PyObject* column = create_column_description(&description->columns[ix]);
+                if (column)
+                {
+                    PyTuple_SET_ITEM(tuple, (Py_ssize_t)ix, column); /* column reference stolen by PyTuple_SET_ITEM */
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            if (PyErr_Occurred())
+            {
+                Py_DECREF(tuple);
+                tuple = NULL;
             }
             else
             {
-                break;
+                description->_obj = tuple;
```

Comment:
I would set `tuple = NULL` and add a comment that the reference was moved to `description->_obj`

Reply:
Done.